src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -566,30 +583,65 @@ export class ResultsViewPageStore
         this.driverAnnotationSettings = buildDriverAnnotationSettings(
             () => store.didHotspotFailInOncoprint
         );
-        this.driverAnnotationsReactionDisposer = reaction(
-            () => this.urlWrapper.query.cancer_study_list,
-            () => {
-                this.driverAnnotationSettings = buildDriverAnnotationSettings(
-                    () => store.didHotspotFailInOncoprint
-                );
-            },
-            { fireImmediately: true }
+
+        this.pageUserSession = new PageUserSession<ResultPageSettings>(
+            appStore,
+            ServerConfigHelpers.sessionServiceIsEnabled()
+        );
+
+        this.pageUserSession.id = {
+            page: PageType.RESULTS_VIEW,
+            origin: this.cancerStudyIds,
+        };
+
+        this.reactionDisposers.push(
+            reaction(
+                () => this.urlWrapper.query.cancer_study_list,
+                () => {
+                    this.driverAnnotationSettings = buildDriverAnnotationSettings(
+                        () => store.didHotspotFailInOncoprint
+                    );
+                },
+                { fireImmediately: true }
+            )
+        );
+
+        this.reactionDisposers.push(
```

Comment:
check to see if you actually get a whole new store every time you requery (change study ids).  if that's the

Reply:
Solved: Does not re-instantiate resultsViewPageStore when we update cancer studies, so this is actually needed